agent/src/util/device_plugin_service.rs

```diff
@@ -751,16 +750,17 @@ pub async fn build_device_plugin(
     shared: bool,
     instance_properties: HashMap<String, String>,
     instance_map: InstanceMap,
+    device_plugin_path: &str,
 ) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
     info!("build_device_plugin - entered for device {}", instance_name);
     let capability_id: String = format!("{}/{}", AKRI_PREFIX, instance_name);
     let unique_time = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH)?;
     let device_endpoint: String = format!("{}-{}.sock", instance_name, unique_time.as_secs());
-    let socket_path: String = format!(
-        "{}{}",
-        DEVICE_PLUGIN_PATH.to_string(),
-        device_endpoint.clone()
-    );
+    let socket_path: String = Path::new(device_plugin_path)
```

Comment:
Why does this need to be a String?

Reply:
Specifically, wouldn't &str be more efficient than cloning String along?

Reply:
Yes, it would, but i think I ran into some lifetime issues that seemed to bubble up to this closure in `serve`: `connect_with_connector(service_fn(move |_: Uri| UnixStream::connect(path.clone())))`. Feel free to give it a go! Would love to know how to make the `&str` lifetime live long enough without declaring it static.

Reply:
Ok, will leave probably for another PR when I get a chance.

Please let me know if anything else looks suspicious, Rust newbie here.